packages/application-shell/src/components/quick-access/quick-access.tsx

```diff
@@ -408,7 +414,7 @@ const QuickAccess = (props: Props) => {
         commands.push({
           id: `go/product-by-key/product(${productId})`,
           text: intl.formatMessage(messages.showProduct, {
-            productName: data.productByKey.key,
+            productName: searchText,
```

Comment:
This was the only place where the `key` was used, but given the query by key, we should be able here to use the `searchText` value (which is the same as the key).